specs/engine.spec/src/main/resources/META-INF/zilla/core.idl

```diff
@@ -108,6 +108,7 @@ scope core
             int64 timestamp;
             int64 traceId;
             int64 namespacedId;
+            octets extension;
```

Comment:
We do this in frames because extension is optional, but it is required for events (agree?) so perhaps it would be more convenient to put `Extension` as the type here instead of `octets`?